File: projects/node-communication/node-writer.js
var zmq = require('zmq');
var writer = zmq.socket('pub');

var local = 'ipc:///tmp/zmq0';

process.stdin.setEncoding('utf8');

writer.bind(local, function(err) {
  if(err)
    console.log(err);
  else
    console.log("Binding to "+local);
});

// Sending stdin to the ipc socket
process.stdin.on('readable', function() {
  var chunk = process.stdin.read();
  if (chunk !== null) {
    writer.send(chunk);
    //console.log("## envoi de "+chunk);
  }
});

File: projects/node-communication/node-writer.py
#!/usr/bin/python

import zmq

# Loading 0MQ context
context = zmq.Context()

# Configuration
local = 'ipc:///tmp/zmq0'

# Local socket
localsock = context.socket(zmq.PUB)
localsock.connect(local)

while True:
	message = raw_input("Sending to "+local+" ")
	if message in ["Q","q"]: 
		print "Closing"
		break
	else:
		localsock.send_multipart(message)
